.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,233 @@
+# Pre-requisites:
+# - AWS Account with correct permissions
+# - `gh-pages` branch
+# - LOGS_NAMESPACE? + LOGS_GROUP
+
+name: Soak Testing
+on:
+  schedule:
+    - cron: '0 */6 * * *' # Every 6 hours
+  # push:
+  #   branches: [ main ]
+env:
+  # Language Specific
+  PROCESS_COMMAND_LINE: /usr/local/bin/python3 application.py
+  PROCESS_EXECUTABLE_NAME: python3
+  # - Every language must also set the path for the sample-app in the `Build and Push Docker image` step.
+  # Global Env Vars
+  AWS_REGION: us-west-2
+  SOAK_TEST_DURATION_IN_MINUTES: 300
+  HOSTMETRICS_COLLECTION_INTERVAL_IN_SECONDS: 600
+  CPU_LOAD_THRESHOLD: 75
+  TOTAL_MEMORY_THRESHOLD: 1610612736
+  # Add this?
+  # RATE: TPS
+
+jobs:
+  test_apps_and_publish_results:
+    name: Publish app and Soak Test - (${{ matrix.app-platform }}, ${{ matrix.instrumentation-type }})
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+      issues: write
+    strategy:
+      fail-fast: false
+      matrix:
+        app-platform: [ flask ]
+        # instrumentation-type: [ auto, manual ]
+        instrumentation-type: [ auto ]
+    env:
+      IMAGE_SUFFIX: sample-app-${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}-${{ github.sha }}
+    steps:
+      - name: Configure Soak Test environment variables
+        # TODO: Does the LOGS_NAMESPACE need a `-$GITHUB_RUN_ID`?
+        run: >-
+          echo "LOGS_NAMESPACE=${{ github.repository }}/soak-tests" >> $GITHUB_ENV;
+          echo "NUM_OF_CPUS=$(nproc --all)" >> $GITHUB_ENV;
+      - uses: actions/checkout@v2
+      - name: Log in to the GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Extract metadata (tags, labels) for Docker
+        id: docker-image-metadata
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ghcr.io/${{ github.repository }}-${{ env.IMAGE_SUFFIX }}
+      # TODO: Could we just build in `docker-compose` build/context without publishing?
```

Comment:
Seems like that's what you're doing now

Reply:
I'm building from a path "context" for the `alarm-poller` and `load-generator`, but I am pulling an image for the `otel-collector-contrib` and the `sample-app`.

What I meant in this comment was that instead of building, publishing, and pulling in an image for the **sample-app**, we could use that `build/context` feature from `docker-compose` to build and use a throw away Docker Image of the Sample App. We can remove all the steps above to build and publish the **sample-app** image. The only downside is we wouldn't have [publicly published versions of the Sample App Docker Image like I do on my repo](https://github.com/NathanielRN?tab=packages&repo_name=aws-otel-python-soaking-tests-demo), but I think that's fine because anyone can go to a commit and build the sample-app at that commit if they want to?

I will also update the comment to say that more explicitly.

Reply:
Any thoughts about this? Whether we can skip publishing the Docker Image?